notebooks/1_maze_tuto.ipynb

```diff
@@ -0,0 +1,740 @@
+    "    x: int\n",
+    "    y: int\n",
+    "\n",
+    "\n",
+    "class Action(Enum):\n",
+    "    up = 0\n",
+    "    down = 1\n",
+    "    left = 2\n",
+    "    right = 3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Domain type\n",
+    "Then we define the domain type from a base template (`DeterministicPlanningDomain`) with optional refinements (`UnrestrictedActions` and `Renderable`). This corresponds to the following characteristics:\n",
+    "- `DeterministicPlanningDomain`:\n",
+    "    - only one agent\n",
+    "    - deterministic starting state\n",
+    "    - handle only actions\n",
+    "    - actions are sequential\n",
+    "    - deterministic transitions\n",
+    "    - white box transition model\n",
+    "    - goal states are defined\n",
+    "    - positive costs (i.e. negative rewards)\n",
+    "    - fully observable\n",
+    "    - renderable (can be displayed)\n",
+    "- `UnrestrictedActions`: all actions are available at each step\n",
+    "- `Renderable`: can be displayed\n",
+    "\n",
+    "We also specify the type of states, observations, events, transition values, ... \n",
+    "\n",
+    "This is needed so that solvers know how to work properly with this domain, and this will also help IDE or Jupyter to propose you intelligent code completion."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class D(DeterministicPlanningDomain, UnrestrictedActions, Renderable):\n",
+    "    T_state = State  # Type of states\n",
+    "    T_observation = State  # Type of observations\n",
+    "    T_event = Action  # Type of events\n",
+    "    T_value = float  # Type of transition values (rewards or costs)\n",
+    "    T_predicate = bool  # Type of logical checks\n",
+    "    T_info = None  # Type of additional information in environment outcome\n",
+    "    T_agent = Union  # Inherited from SingleAgent"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Actual domain class\n",
+    "We can now implement the maze domain by \n",
+    "- deriving from the above domain type\n",
+    "- filling all non-implemented methods \n",
+    "- adding a constructor to define the maze & start/end positions.\n",
+    "\n",
+    "We also define (to help solvers that can make use of it)\n",
+    "- an heuristic for search algorithms\n",
+    "\n",
+    "\n",
+    "*NB: To know the methods not yet implemented, one can either use an IDE which can find them automatically or the [code generators](https://airbus.github.io/scikit-decide/guide/codegen.html) page in the online documentation, which generates the corresponding boilerplate code.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MazeDomain(DeterministicPlanningDomain, UnrestrictedActions, Renderable):\n",
```

Comment:
Should be `class MazeDomain(D)` instead

Reply:
Oops, again a test that remained behind, my bad... Changed!